src/middlewared/middlewared/plugins/pool.py

```diff
@@ -1377,91 +1376,55 @@ async def import_pool(self, job, data):
                 }]
             }
         """
-
-        pool = None
-        for p in await self.middleware.call('zfs.pool.find_import'):
-            if p['guid'] == data['guid']:
-                pool = p
-                break
-        if pool is None:
-            raise CallError(f'Pool with guid "{data["guid"]}" not found', errno.ENOENT)
-
+        # import zpool
         try:
-            job.check_pipe("input")
-            key = job.pipes.input.r
-        except ValueError:
-            key = None
-
-        passfile = None
-        if key and data.get('passphrase'):
-            encrypt = 2
-            passfile = tempfile.mktemp(dir='/tmp/')
-            with open(passfile, 'w') as f:
-                os.chmod(passfile, 0o600)
-                f.write(data['passphrase'])
-        elif key:
-            encrypt = 1
-        else:
-            encrypt = 0
-
-        pool_name = data.get('name') or pool['name']
-        pool_id = None
-        try:
-            pool_id = await self.middleware.call('datastore.insert', 'storage.volume', {
-                'vol_name': pool_name,
-                'vol_encrypt': encrypt,
-                'vol_guid': data['guid'],
-                'vol_encryptkey': str(uuid.uuid4()) if encrypt else '',
-            })
-            pool = await self.middleware.call('pool.query', [('id', '=', pool_id)], {'get': True})
-            if encrypt > 0:
-                if not os.path.exists(GELI_KEYPATH):
-                    os.mkdir(GELI_KEYPATH)
-                with open(pool['encryptkey_path'], 'wb') as f:
-                    f.write(key.read())
-
-            await self.middleware.call('pool.scrub.create', {
-                'pool': pool_id,
-            })
-
-            await self.middleware.call('zfs.pool.import_pool', pool['guid'], {
-                'altroot': '/mnt',
-                'cachefile': ZPOOL_CACHE_FILE,
-            }, True, None, pool_name)
-
-            await self.middleware.call('zfs.dataset.update', pool_name, {
-                'properties': {
-                    'aclinherit': {'value': 'passthrough'},
-                },
-            })
+            guid = data['guid']
+            opts = {'altroot': '/mnt', 'cachefile': ZPOOL_CACHE_FILE}
+            any_host = True
+            use_cachefile = None
+            new_name = data.get('name')
+            await self.middleware.call('zfs.pool.import_pool', guid, opts, any_host, use_cachefile, new_name)
+        except Exception as e:
```

Comment:
I think we should have try/else clause for updating database when we have imported the pool. If any call fails for any reason like `zfs.dataset.update`, we would at least have record of the import in our database

Reply:
Probably better if we insert before importing and removing on exception from database in case pool import but a later step fails in the importing pool method

Reply:
What happens if the zpool was exported via the CLI? When I try to insert first, it will raise a unique constraint traceback. How do we handle that?

Reply:
Is that a behavior we support ? I am sure there can be quite other cases in different parts which can arise to manual intervention ?

Reply:
If we remove the db entry but the zpool import is successful and so we lave the zpool imported, what does that really do for us?

Reply:
My changes still produce the same outcome btw. If importing fails (or anything after it), we won't reach the part of inserting a db entry.

Reply:
I am not sure what aspect you are referring to ? It would probably be nice if you could clarify why we are changing existing behavior ?

Reply:
I'm not changing the behavior? If any API calls fail before we get to the point of inserting the db entry, then the zpool will be left imported. Before my changes, this would also occur.

Reply:
The db entry is made first and then the pool is imported and on failure the entry is removed. Is that still happening ?

Reply:
The end-result is the same. If anything fails during import process, no db entry is made.